packages/patrol/lib/src/extensions.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:meta/meta.dart';
+// ignore: implementation_imports
+import 'package:test_api/src/backend/invoker.dart';
+
+/// Provides convenience methods for [Invoker].
+@internal
+extension InvokerX on Invoker {
+  /// Returns the full name of the current test (names of all ancestor groups +
+  /// name of the current test).
+  String fullCurrentTestName() {
+    final parentGroupName = liveTest.groups.last.name;
```

Comment:
What if there are no groups?

Reply:
(Assuming no fatal programming error in Patrol) 

Every `test()` always has at least one parent group - the one in `test_bundle.dart`:

<details><summary>sample</summary>

```dart
  // START: GENERATED TEST GROUPS
  group('android_app_test', android_app_test.main);
  group('example_test', example_test.main);
  group('open_app_test', open_app_test.main);
  group('open_quick_settings_test', open_quick_settings_test.main);
  group('permissions.notifications_test', permissions__notifications_test.main);
  group('permissions.permissions_location_test', permissions__permissions_location_test.main);
  group('permissions.permissions_many_test', permissions__permissions_many_test.main);
  group('service_airplane_mode_test', service_airplane_mode_test.main);
  group('service_bluetooth_test', service_bluetooth_test.main);
  group('service_cellular_test', service_cellular_test.main);
  group('service_dark_mode_test', service_dark_mode_test.main);
  group('service_wifi_test', service_wifi_test.main);
  group('swipe_test', swipe_test.main);
  group('webview_hackernews_test', webview_hackernews_test.main);
  group('webview_leancode_test', webview_leancode_test.main);
  group('webview_stackoverflow_test', webview_stackoverflow_test.main);
  // END: GENERATED TEST GROUPS
```

</details>